pkg/downwardmetrics/virtio-serial/server.go

```diff
@@ -129,6 +130,8 @@ func (s *downwardMetricsServer) serve(ctx context.Context, conn net.Conn) {
 	// is canceled
 	newRequest := make(chan reqResult, 1)
 	reader := bufio.NewReader(conn)
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	defer cancelCtx()
 
 	for {
```

Comment:
Do we really need to spawn a new routine in every iteration of the loop?

Reply:
@germag Would you mind to take this one please?